### PR TITLE
[FEAT] [SECURITY] OAuth component

### PR DESCRIPTION
**Pull request by Guikingone**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 ?
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no    
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | To add

Hi everyone 👋 

Here's my first "big" PR on Symfony, recently, I've decided to implement OAuth authentication in a side-project application, in order to do it, I've used `HttpClient` (thanks to Nicolas Grekas) and I've been amazed by the simplicity of the implementation but something was weird ... Why Symfony doesn't have a "native" OAuth support/Client/Server/Component? 
Then I saw that the hackaton highlighted the idea of having a native support: https://github.com/symfony/symfony/issues/30914

So, I've decided to take time to work on a strict implementation (well, as strict as I could do) thanks to the RFC and `HttpClient` and here's the first result, a new "OAuth component" (actually, that's an experimentation 😄), for now, this experimentation only support the "client" aspect (the server has been started) and that I'm aware that a lot of feature can be lacking or to update. 

So, why opening this PR now ? 

First, I'm aware that this is a WIP and that a lot of feedback can be linked to some choices that I've made, please, do not hesitate to explain me if there's a better option/way to do it.

Second, I think that it's a good idea to discuss about this idea and explain why it's not a good idea/option to do it as a component (or even as a feature?) before going further on the development 🙂 

So here's the PR with the first parts of the components, feel free to comments and have a great day 🙂

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
should target 4.4, master is only for deprecated code removals :)

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -883,6 +885,82 @@ private function addCacheSection(ArrayNodeDefinition $rootNode)
...
+                                ->scalarNode('client_secret')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('authorization_url')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('redirect_uri')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('accessToken_url')
```
Comment (Koc) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: mixed camelCase with underscore

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -883,6 +885,82 @@ private function addCacheSection(ArrayNodeDefinition $rootNode)
...
+                                ->end()
+                                ->scalarNode('authorization_url')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('redirect_uri')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('access_token_url')
+                                    ->isRequired()
+                                ->end()
```
Comment (Taluu) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: imo most of these options shouldn't have a "default", as for a client everything should be specific (e.g I shouldn't have the same key for multiple providers).

Especially as all of these (or almost) are required for each providers

Comment (Guikingone) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Thanks, I'm gonna change this 🙂

**Comment by Guikingone** [comment]
Hi everyone 👋 

This PR is closed, I've create a new one (wrong rebase process on this one 🙁 ) -> https://github.com/symfony/symfony/pull/31952

Sorry for the new one :(

**Pull request closed without merging**